prov/sm2/src/sm2_msg.c

```diff
@@ -38,8 +38,34 @@
 #include "ofi_iov.h"
 #include "sm2.h"
 
-static inline int sm2_select_proto(uint64_t total_len)
+static inline int sm2_select_proto(void **desc, size_t iov_count,
+				   uint64_t op_flags, uint64_t total_len)
 {
+	struct ofi_mr *sm2_desc;
+	enum fi_hmem_iface iface;
+
+	if (desc && *desc) {
+		sm2_desc = (struct ofi_mr *) *desc;
+		iface = sm2_desc->iface;
+		if (iface == FI_HMEM_CUDA) {
```

Comment:
I think we should add an else with a warn printout "Non CUDA devices not supported"

Reply:
Host memory can also have memory descriptors. So we would need to add another check for FI_HMEM_SYSTEM and that doesn't seem worth it to me.